src/Parser/Mage/Frost/Modules/Features/BrainFreeze.js

```diff
@@ -11,81 +11,111 @@ const debug = false;
 // Brain Freeze appears to always fall after Flurry cast, but not always on same timestamp. Giving a margin here.
 const PROC_WINDOW_MS = 100;
 
+// When Glacial Spike is talented, it's OK to overwrite Brain Freeze procs if you have at least this many icicles *during* the frost bolt cast
+const ICICLE_BREAKPOINT = 3;
+
 class BrainFreeze extends Analyzer {
-	lastFlurryTimestamp;
+  lastFlurryTimestamp;
 
-	overwrittenProcs = 0;
-	expiredProcs = 0;
-	totalProcs = 0;
-	flurryWithoutProc = 0;
+  overwrittenProcs = 0;
+  okOverwrittenProcs = 0;
+  expiredProcs = 0;
+  totalProcs = 0;
+  flurryWithoutProc = 0;
 
-	on_byPlayer_applybuff(event) {
-		const spellId = event.ability.guid;
-    if (spellId !== SPELLS.BRAIN_FREEZE.id) {
-			return;
-		}
-		this.totalProcs += 1;
-	}
+  // Tracks whether the last brain freeze generator to be cast was Ebonbolt or Frostbolt
+  wasLastGeneratorEB = false;
+
+  constructor(...args) {
+    super(...args);
+    this.glacialSpikeTalented = this.selectedCombatant.hasTalent(SPELLS.GLACIAL_SPIKE_TALENT.id);
+  }
 
-	on_byPlayer_refreshbuff(event) {
-		const spellId = event.ability.guid;
+  on_byPlayer_applybuff(event) {
+    const spellId = event.ability.guid;
+    if (spellId === SPELLS.BRAIN_FREEZE.id) {
+      this.totalProcs += 1;
+    }
+  }
+
+  on_byPlayer_refreshbuff(event) {
+    const spellId = event.ability.guid;
     if (spellId !== SPELLS.BRAIN_FREEZE.id) {
-			return;
-		}
-		this.overwrittenProcs += 1;
-		this.totalProcs += 1;
-    debug && console.log("Brain Freeze proc overwritten @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+      return;
+    }
+    this.totalProcs += 1;
+
+    if (!this.glacialSpikeTalented || this.wasLastGeneratorEB) {
+      this.overwrittenProcs += 1;
+      debug && console.log("Brain Freeze proc overwritten w/o GS talented or by EB @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+      return;
+    }
+
+    // Get the number of icicles the player had *after* the cast of the frostbolt
+    const stacks = this.selectedCombatant.getBuff(SPELLS.ICICLES_BUFF.id).stacks || 0;
+
+    // The breakpoint is based on the number of icicles the player had *during* the cast of frostbolt, so increment by 1 to check against current icicles
+    if (stacks < ICICLE_BREAKPOINT + 1) {
+      this.overwrittenProcs += 1;
+      debug && console.log("Brain Freeze proc overwritten w/ GS talented with too few icicles @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+    } else {
+      this.okOverwrittenProcs += 1;
+      debug && console.log("Acceptable Brain Freeze proc overwritten w/ GS talented @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+    }
   }
 
-	on_byPlayer_cast(event) {
-		const spellId = event.ability.guid;
+  on_byPlayer_cast(event) {
+    const spellId = event.ability.guid;
+    if(spellId === SPELLS.FROSTBOLT.id || spellId === SPELLS.EBONBOLT_TALENT.id) {
```

Comment:
I believe the prefered style is to return on any spells that are not the ones you are looking for

i.e. 
if (spellId !== SPELLS.FROSTBOLT.id && spellId !== SPELLS.EBONBOLT_TALENT.id) {
    return;
}

I only mention it cause im sure someone else will when they go to approve it.

Reply:
That doesn't work when you have two different conditionals though. It's also checking for flurry in there with two different paths. If I return early not matching FB or EB then the flurry conditional will never get checked

Reply:
good to know. I only suggested it cause it has been added as a suggestion on my PRs before so i figured someone would say it